src/contrib/cluster/Akka.DistributedData/Replicator.cs

```diff
@@ -479,35 +486,91 @@ private void Load()
             Receive(new Action<Write>(IgnoreDebug));
             Receive(new Action<Status>(IgnoreDebug));
             Receive(new Action<Gossip>(IgnoreDebug));
+
+            Receive<ClusterEvent.IClusterDomainEvent>(msg =>
+            {
+                if(!NormalReceive(msg))
+                    Unhandled(msg);
+            });
+
+            ReceiveAny(_ => Stash.Stash());
         }
 
-        private void NormalReceive()
+        private bool NormalReceive(object message)
         {
-            Receive<Get>(g => ReceiveGet(g.Key, g.Consistency, g.Request));
-            Receive<Update>(msg => ReceiveUpdate(msg.Key, msg.Modify, msg.Consistency, msg.Request));
-            Receive<Read>(r => ReceiveRead(r.Key));
-            Receive<Write>(w => ReceiveWrite(w.Key, w.Envelope));
-            Receive<ReadRepair>(rr => ReceiveReadRepair(rr.Key, rr.Envelope));
-            Receive<DeltaPropagation>(msg => ReceiveDeltaPropagation(msg.FromNode, msg.ShouldReply, msg.Deltas));
-            Receive<FlushChanges>(_ => ReceiveFlushChanges());
-            Receive<DeltaPropagationTick>(_ => ReceiveDeltaPropagationTick());
-            Receive<GossipTick>(_ => ReceiveGossipTick());
-            Receive<ClockTick>(c => ReceiveClockTick());
-            Receive<Internal.Status>(s => ReceiveStatus(s.Digests, s.Chunk, s.TotalChunks));
-            Receive<Gossip>(g => ReceiveGossip(g.UpdatedData, g.SendBack));
-            Receive<Subscribe>(s => ReceiveSubscribe(s.Key, s.Subscriber));
-            Receive<Unsubscribe>(u => ReceiveUnsubscribe(u.Key, u.Subscriber));
-            Receive<Terminated>(t => ReceiveTerminated(t.ActorRef));
-            Receive<ClusterEvent.MemberWeaklyUp>(m => ReceiveMemberWeaklyUp(m.Member));
-            Receive<ClusterEvent.MemberUp>(m => ReceiveMemberUp(m.Member));
-            Receive<ClusterEvent.MemberRemoved>(m => ReceiveMemberRemoved(m.Member));
-            Receive<ClusterEvent.IMemberEvent>(m => ReceiveOtherMemberEvent(m.Member));
-            Receive<ClusterEvent.UnreachableMember>(u => ReceiveUnreachable(u.Member));
-            Receive<ClusterEvent.ReachableMember>(r => ReceiveReachable(r.Member));
-            Receive<GetKeyIds>(_ => ReceiveGetKeyIds());
-            Receive<Delete>(d => ReceiveDelete(d.Key, d.Consistency, d.Request));
-            Receive<RemovedNodePruningTick>(r => ReceiveRemovedNodePruningTick());
-            Receive<GetReplicaCount>(_ => ReceiveGetReplicaCount());
+            switch (message)
+            {
+                case IDestinationSystemUid msg:
+                    if (msg.ToSystemUid.HasValue && msg.ToSystemUid != _selfUniqueAddress.Uid)
```

Comment:
Special case for `Status` and `Gossip` messages, we filter out messages that are not intended for this specific node. Processing these messages can poison the whole network with unprocessable messages that ends up being dead lettered.

Reply:
Nice catch - good to know.